internal/usecase/storage/filestore/filestore.go

```diff
@@ -0,0 +1,78 @@
+package filestore
+
+import (
+	"bufio"
+	"bytes"
+	"encoding/json"
+	"github.com/Yasuhiro-gh/url-shortener/internal/config"
+	"github.com/Yasuhiro-gh/url-shortener/internal/usecase/storage"
+	"os"
+)
+
+var IDCounter int
+
+func CreateFileStorage() error {
+	_, err := os.OpenFile(config.Options.FileStoragePath, os.O_RDWR|os.O_CREATE|os.O_APPEND, 0666)
+	return err
+}
+
+type Record struct {
+	ID          int    `json:"uuid"`
+	ShortURL    string `json:"short_url"`
+	OriginalURL string `json:"original_url"`
+}
+
+func MakeRecord(shortURL, originalURL string) error {
+	r := Record{ID: IDCounter + 1, ShortURL: shortURL, OriginalURL: originalURL}
+
+	rm, err := json.Marshal(r)
+	if err != nil {
+		return err
+	}
+
+	file, err := os.OpenFile(config.Options.FileStoragePath, os.O_RDWR|os.O_CREATE|os.O_APPEND, 0777)
+	if err != nil {
+		return err
+	}
+	defer file.Close()
+
+	_, err = file.Write(rm)
+	if err != nil {
+		return err
+	}
+
+	_, err = file.Write([]byte("\n"))
```

Comment:
```suggestion
	_, err = file.WriteString("\n")
```

Это так, чуть позанудствовать - более быстрая версия (под капоотом unsafe)